README.md

```diff
@@ -51,10 +51,12 @@ Note that the `*_path` elements are for very specific use cases where one might
 &nbsp;
 
 ## `restapi` datasource configuration
+Note that you must set the 'id_attribute' key on the restapi provider in order to use this datasource since it is used to set the ID of the discovered object that matches the search criteria.
 - `path` (string, required): The API path on top of the base URL set in the provider that represents objects of this type on the API server.
-- `search_key` (string, required): When reading search results from the API, this key is used to identify the specific record to read. This should be a unique record such as 'name'.
-- `search_value` (string, required): The value of 'search_key' will be compared to this value to determine if the correct object was found. Example: if 'search_key' is 'name' and 'search_value' is 'foo', the record in the array returned by the API with name=foo will be used.
-- `results_key` (string, required): When issuing a GET to the path, this JSON key is used to locate the results array. The format is 'field/field/field'. Example: 'results/values'. If omitted, it is assumed the results coming back are to be used exactly as-is.
+- `search_key` (string, required): When reading search results from the API, this key is used to identify the specific record to read. This should be a unique field within each record such as 'name' or a path to such an field in the form 'field/field/field'. Example: 'attributes/name' would search for the name field under the attributes map of each record found within the portion of the API response indicated by 'results_key'.
```

Comment:
Thank you for clarifying the intent - this is very helpful. Can you please copy/paste these same descriptions into the datasource field descriptions to keep README and the datasource in sync?

Reply:
Yes.  I just did new commit in which I synced all the descriptions in datasource_api_object.go to match what is in the README.md since I think the latter are better.

Note that debug on the resource also has to be updated to match README.md.  I did not want to do that on this PR, but I could if you want me to.

Reply:
Actually, I went ahead and changed debug for the resource both in resource_api_object.go and README.md to match what I had added for debug in datasource_api_object.go

Reply:
Thanks!